rd_ui/app/scripts/controllers/alerts.js

```diff
@@ -120,6 +122,40 @@
       },
       controller: function ($scope) {
         $scope.subscribers = AlertSubscription.query({alertId: $scope.alertId});
+        $scope.subscription = {};
+        $scope.destinations = Destination.query();
+
+        $scope.destinationsDisplay = function(destination) {
+            var icons = {
+                'slack': 'fa-slack',
+                'email': 'fa-envelope',
```

Comment:
It will be defined in `BaseDestination` derived classes, a method that returns the icon. And it will be returned in the API calls both for types and destination instances.